package/src/method-element.js

```diff
@@ -146,6 +154,29 @@ class MethodElement {
     return data
   }
 
+  /**
+   * Prepare data to handle each type of intent differently.
+   */
+  intentData () {
+    const name = camelCase(this.settings.intent_type)
+    let handler = 'confirm'
+    let data = { payment_method: this.extraData() }
+    if (this.settings.intent_methods.includes('sepa_debit')) {
+      handler += 'SepaDebit'
```

Comment:
I’d prefer to have the full handler name here (ie. `'confirmSepaDebit'`). Makes grepping easier ;-) … In general I think a mapping from `this.settings.intent_type` to the appropriate `name` and `handler` would be a better choice than relying on some pattern in the stripe API’s method names.

Reply:
Tried to make it a bit more transparent! Having the full handler name is IMHO neither feasable nor easier to read: the first part is defined by the intent method, the second part by the intent type, so each new method would require the same nested if to choose between it’s 2 handlers… (If Stripe.js changes that, we’d have to change our code anyway)

As for the name, that actual *is* the camel case version of the type passed by php, can’t make that more explicit.

Reply:
Could you maybe include a link to appropriate page in the stripe docs? I’m having a hard time guessing what the possible values are, and how many different paths the code could go on from there.

Reply:
There is no doc listing *all* handlers, they're all in separate chapters of the doc…

Reply:
Ok, then there is nothing in the docs that describes how the handler-name should be created?

Reply:
Turns out there are only 5 possible handlers (that I could find in the current doc) anyway: 

* confirmCardPayment
* confirmSepaDebitPayment
* confirmIdealPayment
* confirmCardSetup
* confirmSepaDebitSetup

→ [amended commit](https://github.com/moreonion/stripe_payment/pull/9/commits/9efa3706af482df1108f5178e32ea2f9e7dddf3b)